### ISSUE TITLE
Promise is never resolved when a server is down

### ISSUE THREAD

**Issue by igor-lemon** [parse-community/Parse-SDK-JS#1417]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse JS SDK](https://github.com/parse-community/Parse-SDK-JS/releases). 

### Issue Description
The Promise from RESTController will never resolved when a server is down. 

### Steps to reproduce
Set some wrong URL as parse server url , `parse.serverURL = 'https://some-wrong-site.com'`
Call
```js
 const parseConfig = await Parse.Config.get()
```

### Actual Outcome
The promise will never resolve.

### Expected Outcome
The promise should be rejected.

###  Environment
Client
- Parse JS SDK version: `3.3.1`

**Comment by mtrezza** [comment]
What would be the expectation? To reject the promise after a timeout?

**Comment by igor-lemon** [comment]
Yep, I think we should make a few retries (if I remember 5 by default) but after the promise should be rejected. 
We have infinity waiting at the moment. 

I think it related with `if (xhr.readyState !== 4 || handled || xhr._aborted)` condition. 
We have `xhr.readyState` === `1` and just exit from the handler and after we don't make any repeat attempts.
I solved it at the local build when reject the promise at `RESTControlle.js`

```js
 if (xhr.readyState !== 4 || handled || xhr._aborted) {
   promise.reject(xhr);
 }
```

**Comment by mtrezza** [comment]
Sounds reasonable. Do you think you can submit a PR with a test and a fix for this?

**Comment by igor-lemon** [comment]
Yep. I'll try to do it today

**Comment by igor-lemon** [comment]
Hi, @mtrezza! 
Will you check the PR?

**Comment by mtrezza** [comment]
Sure, feel free to request a review inside the PR anytime.

**Comment by dplewis** [comment]
Closing as resolved via https://github.com/parse-community/Parse-SDK-JS/pull/1418#issuecomment-950868778
